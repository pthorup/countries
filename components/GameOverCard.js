import styles from '../styles/GameOverCard.module.css'

const GameOverCard = ({ lives, onRandomCountry, answer, isGameOver }) => {
   return (
      <>
         {lives === 0 ? (
            <div className={styles.gameLost}>
               <div>
                  <p>Sorry, you ran out of lives.</p>
                  <p>
                     Correct answer is:{' '}
                     <span className={styles.correctAnswer}>{answer}</span>{' '}
                  </p>
               </div>
               <button
                  className={styles.playAgainBtn}
                  onClick={onRandomCountry}
               >
                  Play again
               </button>
            </div>
         ) : isGameOver ? (
            <div className={styles.gameWon}>
               <div>You won!!!</div>
               <button
                  className={styles.playAgainBtn}
                  onClick={onRandomCountry}
               >
                  Play again
               </button>
            </div>
         ) : (
            ''
         )}
      </>
   )
}

export default GameOverCard
