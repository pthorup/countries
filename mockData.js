const answer = 'Sweden'

const continent = {
   name: 'Africa',
   countries: [
      {
         __typename: 'Country',
         name: 'Angola',
         code: 'AO',
      },
      {
         __typename: 'Country',
         name: 'Burkina Faso',
         code: 'BF',
      },
      {
         __typename: 'Country',
         name: 'Burundi',
         code: 'BI',
      },
      {
         __typename: 'Country',
         name: 'Benin',
         code: 'BJ',
      },
      {
         __typename: 'Country',
         name: 'Botswana',
         code: 'BW',
      },
      {
         __typename: 'Country',
         name: 'Democratic Republic of the Congo',
         code: 'CD',
      },
      {
         __typename: 'Country',
         name: 'Central African Republic',
         code: 'CF',
      },
      {
         __typename: 'Country',
         name: 'Republic of the Congo',
         code: 'CG',
      },
      {
         __typename: 'Country',
         name: 'Ivory Coast',
         code: 'CI',
      },
      {
         __typename: 'Country',
         name: 'Cameroon',
         code: 'CM',
      },
      {
         __typename: 'Country',
         name: 'Cape Verde',
         code: 'CV',
      },
      {
         __typename: 'Country',
         name: 'Djibouti',
         code: 'DJ',
      },
      {
         __typename: 'Country',
         name: 'Algeria',
         code: 'DZ',
      },
      {
         __typename: 'Country',
         name: 'Egypt',
         code: 'EG',
      },
      {
         __typename: 'Country',
         name: 'Western Sahara',
         code: 'EH',
      },
      {
         __typename: 'Country',
         name: 'Eritrea',
         code: 'ER',
      },
      {
         __typename: 'Country',
         name: 'Ethiopia',
         code: 'ET',
      },
      {
         __typename: 'Country',
         name: 'Gabon',
         code: 'GA',
      },
      {
         __typename: 'Country',
         name: 'Ghana',
         code: 'GH',
      },
      {
         __typename: 'Country',
         name: 'Gambia',
         code: 'GM',
      },
      {
         __typename: 'Country',
         name: 'Guinea',
         code: 'GN',
      },
      {
         __typename: 'Country',
         name: 'Equatorial Guinea',
         code: 'GQ',
      },
      {
         __typename: 'Country',
         name: 'Guinea-Bissau',
         code: 'GW',
      },
      {
         __typename: 'Country',
         name: 'Kenya',
         code: 'KE',
      },
      {
         __typename: 'Country',
         name: 'Comoros',
         code: 'KM',
      },
      {
         __typename: 'Country',
         name: 'Liberia',
         code: 'LR',
      },
      {
         __typename: 'Country',
         name: 'Lesotho',
         code: 'LS',
      },
      {
         __typename: 'Country',
         name: 'Libya',
         code: 'LY',
      },
      {
         __typename: 'Country',
         name: 'Morocco',
         code: 'MA',
      },
      {
         __typename: 'Country',
         name: 'Madagascar',
         code: 'MG',
      },
      {
         __typename: 'Country',
         name: 'Mali',
         code: 'ML',
      },
      {
         __typename: 'Country',
         name: 'Mauritania',
         code: 'MR',
      },
      {
         __typename: 'Country',
         name: 'Mauritius',
         code: 'MU',
      },
      {
         __typename: 'Country',
         name: 'Malawi',
         code: 'MW',
      },
      {
         __typename: 'Country',
         name: 'Mozambique',
         code: 'MZ',
      },
      {
         __typename: 'Country',
         name: 'Namibia',
         code: 'NA',
      },
      {
         __typename: 'Country',
         name: 'Niger',
         code: 'NE',
      },
      {
         __typename: 'Country',
         name: 'Nigeria',
         code: 'NG',
      },
      {
         __typename: 'Country',
         name: 'Réunion',
         code: 'RE',
      },
      {
         __typename: 'Country',
         name: 'Rwanda',
         code: 'RW',
      },
      {
         __typename: 'Country',
         name: 'Seychelles',
         code: 'SC',
      },
      {
         __typename: 'Country',
         name: 'Sudan',
         code: 'SD',
      },
      {
         __typename: 'Country',
         name: 'Saint Helena',
         code: 'SH',
      },
      {
         __typename: 'Country',
         name: 'Sierra Leone',
         code: 'SL',
      },
      {
         __typename: 'Country',
         name: 'Senegal',
         code: 'SN',
      },
      {
         __typename: 'Country',
         name: 'Somalia',
         code: 'SO',
      },
      {
         __typename: 'Country',
         name: 'South Sudan',
         code: 'SS',
      },
      {
         __typename: 'Country',
         name: 'São Tomé and Príncipe',
         code: 'ST',
      },
      {
         __typename: 'Country',
         name: 'Swaziland',
         code: 'SZ',
      },
      {
         __typename: 'Country',
         name: 'Chad',
         code: 'TD',
      },
      {
         __typename: 'Country',
         name: 'Togo',
         code: 'TG',
      },
      {
         __typename: 'Country',
         name: 'Tunisia',
         code: 'TN',
      },
      {
         __typename: 'Country',
         name: 'Tanzania',
         code: 'TZ',
      },
      {
         __typename: 'Country',
         name: 'Uganda',
         code: 'UG',
      },
      {
         __typename: 'Country',
         name: 'Mayotte',
         code: 'YT',
      },
      {
         __typename: 'Country',
         name: 'South Africa',
         code: 'ZA',
      },
      {
         __typename: 'Country',
         name: 'Zambia',
         code: 'ZM',
      },
      {
         __typename: 'Country',
         name: 'Zimbabwe',
         code: 'ZW',
      },
   ],
}

module.exports = {
   answer,
   continent,
}
