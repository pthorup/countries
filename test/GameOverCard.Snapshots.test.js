import React from 'react'
import GameOverCard from '../components/GameOverCard'
import renderer from 'react-test-renderer'
import { answer } from '../mockData'

it('when lives are over', () => {
   const tree = renderer.create(
      <GameOverCard
         lives={0}
         onRandomCountry={jest.fn()}
         answer={answer}
         isGameOver={false}
      />
   )

   expect(tree).toMatchSnapshot()
})

it('when isGameOver is true', () => {
   const tree = renderer.create(
      <GameOverCard
         lives={3}
         onRandomCountry={jest.fn()}
         answer={answer}
         isGameOver
      />
   )

   expect(tree).toMatchSnapshot()
})
