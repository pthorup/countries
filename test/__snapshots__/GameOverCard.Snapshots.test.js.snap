// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when isGameOver is true 1`] = `
<div
  className="gameWon"
>
  <div>
    You won!!!
  </div>
  <button
    className="playAgainBtn"
    onClick={[MockFunction]}
  >
    Play again
  </button>
</div>
`;

exports[`when lives are over 1`] = `
<div
  className="gameLost"
>
  <div>
    <p>
      Sorry, you ran out of lives.
    </p>
    <p>
      Correct answer is:
       
      <span
        className="correctAnswer"
      >
        Sweden
      </span>
       
    </p>
  </div>
  <button
    className="playAgainBtn"
    onClick={[MockFunction]}
  >
    Play again
  </button>
</div>
`;
